mutiny-core/src/wallet.rs

```diff
@@ -69,37 +68,59 @@ impl MutinyWallet {
             info!("Updated cached fees!");
         });
 
-        maybe_await!(wallet.sync(&self.blockchain, SyncOptions::default()))?;
+        // get first wallet lock that only needs to read
+        let (checkpoints, spks) = {
+            let wallet = self.wallet.try_read()?;
+            let checkpoints = wallet.checkpoints();
+            let spks = wallet
+                .spks_of_all_keychains()
+                .into_iter()
+                .map(|(k, spks)| (k, spks))
+                .collect();
+
+            (checkpoints.clone(), spks)
+        };
+
+        let update = self
+            .blockchain
+            .scan(
+                &checkpoints,
+                spks,
+                core::iter::empty(),
+                core::iter::empty(),
+                50,
+                5,
+            )
+            .await?;
+        // get new wallet lock for writing
+        let mut wallet = self.wallet.try_write()?;
+        wallet.apply_update(update)?;
+        wallet.commit()?;
 
         Ok(())
     }
 
-    pub async fn list_utxos(&self) -> Result<Vec<LocalUtxo>, MutinyError> {
-        Ok(self.wallet.lock().await.list_unspent()?)
+    pub fn list_utxos(&self) -> Result<Vec<LocalUtxo>, MutinyError> {
+        Ok(self.wallet.try_read()?.list_unspent())
     }
```

Comment:
I see a lot of removals of `async`. Is BDK Core now keeping stuff in memory so it doesn't need to read or write directly from indexdb?

Reply:
Yes exactly